java/deepgramLanguageDetection/Index.java

```diff
@@ -0,0 +1,172 @@
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+
+import com.google.gson.Gson;
+import org.apache.commons.validator.routines.UrlValidator;
+final Gson gson=new Gson();
+
+public RuntimeResponse main(RuntimeRequest req,RuntimeResponse res)throws Exception{
+        // Validate that values present in the request are not empty (payload, variables)
+        RuntimeResponse errorResponse=checkEmptyPayloadAndVariables(req,res);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        // Validate the requested payload (URL)
+        String payloadString=req.getPayload().toString();
+        Map<String, Object> payload=gson.fromJson(payloadString,Map.class);
+
+        errorResponse=validatePayload(payload,res);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        // Get file url from payload
+        String fileurl=payload.get("fileUrl").toString();
+
+        // Name API key is not empty
+        String apiKeyVariable="DEEPGRAM_SECRET_KEY";
+
+        errorResponse=checkEmptyAPIKey(req,res,apiKeyVariable);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        String apiKey=req.getVariables().get(apiKeyVariable).toString();
+
+        String deepgramData="";
+        Map<String, Object> responseData=new HashMap<>();
+
+        try{
+            deepgramData=detectLanguage(fileurl,apiKey);
+        }catch(Exception e){
+            responseData.put("success",false);
+            responseData.put("message","Something went wrong while generating the summary, please check with the developers. Error: "+e.getMessage());//TODO
+        return res.json(responseData);
+        }
+        responseData.put("success",true);
+        responseData.put("deepgramData",deepgramData);
+
+        return res.json(responseData);
+}
+
+
+/**
+ * This method will send a POST request to the specified endpoint and return the Deepgram's response
+ *
+ * @param requestBody is the Request Body for the POST request containing the URL to the wav file to Summarize
+ * @param apiKey is the access token used by Deepgram to summarize
+ * @return the Deepgram's response to the POST request
+ * @throws Exception in case of malformed URL, I/O exception, etc.
+ */
+private String detectLanguage(String requestBody,String apiKey)throws Exception{
+  
+        String endpointUrl="https://api.deepgram.com/v1/listen?detect_language=true&punctuate=true";
+        URL url=new URL(endpointUrl);
+        HttpURLConnection con=(HttpURLConnection)url.openConnection();
+
+        con.setRequestMethod("POST");
+        con.setRequestProperty("Content-Type","application/json");
+        con.setRequestProperty("Authorization","Token "+apiKey);
+
+        con.setDoOutput(true);
+
+        //prepare request body
+        requestBody="{"url":"requestBody"}";
```

Comment:
```suggestion
        requestBody="{\"url\":\"requestBody\"}";
```

The escape sequence is necessary to add `"` inside a string

Reply:
My curl request has stopped working, does this change require a change in current structure of payload?
```bash
curl --location --request POST 'http://localhost:3000/' \
--header 'X-Internal-Challenge: secret-key' \
--header 'Content-Type: application/json' \
--data-raw '{"payload": "{\"fileUrl\": \"https://static.deepgram.com/examples/interview_speech-analytics.wav\"}", 
"variables": {"DEEPGRAM_SECRET_KEY":"<YOUR_API_KEY>"}}
'
```